torch_xla/csrc/tensor.cpp

```diff
@@ -699,6 +703,13 @@ void XLATensor::SetInPlaceIrValue(torch::lazy::Value ir_value) {
 }
 
 void XLATensor::AssignIrValue(torch::lazy::Value ir_value) const {
+  // Sharding annotation should be persisted, since the xla_data() is sharded.
```

Comment:
I don't fully get this part, can you explain a bit more. In what case `sharding` will be non-empty? What if it is already being set?

Reply:
Sorry for the confusion, rephrasing:
```
// Sharding annotation it not null, if xla_data() is sharded.
// TODO(yeounoh): Sharding annotation must be removed by explicit call to
// ClearSharding.
  ```